presto-main/src/test/java/com/facebook/presto/operator/TestOperationTimer.java

```diff
@@ -0,0 +1,177 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import com.facebook.presto.operator.OperationTimer.OperationTiming;
+import org.testng.annotations.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testng.Assert.assertEquals;
+
+public class TestOperationTimer
+{
+    public static int blackHole;
+
+    @Test
+    public void testOverallCpuTrackingEnabled()
+    {
+        OperationTiming timing = new OperationTiming();
+
+        OperationTimer timer = new OperationTimer(true, false);
+        doSomething();
+        timer.end(timing);
+
+        assertEquals(timing.getCalls(), 1);
+        assertThat(timing.getCpuNanos()).isGreaterThan(0);
+        assertThat(timing.getWallNanos()).isGreaterThan(timing.getCpuNanos());
+
+        long previousCpuNanos = timing.getCpuNanos();
+        long previousWallNanos = timing.getWallNanos();
+
+        timer = new OperationTimer(true, false);
+        doSomething();
+        timer.end(timing);
+
+        assertEquals(timing.getCalls(), 2);
+        assertThat(timing.getCpuNanos()).isGreaterThan(previousCpuNanos);
+        assertThat(timing.getWallNanos()).isGreaterThan(previousWallNanos);
+    }
+
+    @Test
+    public void testOverallCpuTrackingDisabled()
+    {
+        OperationTiming timing = new OperationTiming();
+
+        OperationTimer timer = new OperationTimer(false, false);
+        doSomething();
+        timer.end(timing);
+
+        assertEquals(timing.getCalls(), 1);
+        assertEquals(timing.getCpuNanos(), 0);
+        assertThat(timing.getWallNanos()).isGreaterThan(0);
+    }
+
+    @Test
+    public void testOperationCpuTrackingEnabled()
+    {
+        OperationTiming overallTiming = new OperationTiming();
+
+        OperationTiming operationTiming1 = new OperationTiming();
+        OperationTiming operationTiming2 = new OperationTiming();
+        OperationTiming operationTiming3 = new OperationTiming();
+
+        OperationTimer timer = new OperationTimer(true, true);
+
+        doSomething();
+        timer.recordOperationComplete(operationTiming1);
+        assertEquals(operationTiming1.getCalls(), 1);
+        assertThat(operationTiming1.getCpuNanos()).isGreaterThan(0);
+        assertThat(operationTiming1.getWallNanos()).isGreaterThan(operationTiming1.getCpuNanos());
+        long previousOperation1CpuTime = operationTiming1.getCpuNanos();
+        long previousOperation1WallTime = operationTiming1.getWallNanos();
+
+        doSomething();
+        timer.recordOperationComplete(operationTiming1);
```

Comment:
Another idea, to simplify things.  Wrap the "OperationTiming" with a nested class that has the "pervious" variables and expected count.  Then you can encapsulate and reuse the validation logic.